tests/tests_app/runners/test_cloud.py

```diff
@@ -92,75 +94,99 @@ def get_cloud_runtime_request_body(**kwargs) -> "Body8":
-        monkeypatch.setattr(backends, "CloudBackend", mock.MagicMock(return_value=cloud_backend))
 
         app = mock.MagicMock()
         app.flows = []
         app.frontend = {}
 
         existing_instance = MagicMock()
         existing_instance.status.phase = V1LightningappInstanceState.STOPPED
-        existing_instance.spec.cluster_id = original_cluster
+        existing_instance.spec.cluster_id = old_cluster or DEFAULT_CLUSTER
         mock_client.lightningapp_instance_service_list_lightningapp_instances.return_value = (
             V1ListLightningappInstancesResponse(lightningapps=[existing_instance])
         )
 
         cloud_runtime = cloud.CloudRuntime(app=app, entrypoint_file="entrypoint.py")
         cloud_runtime._check_uploaded_folder = mock.MagicMock()
 
-        # without requirements file
-        # setting is_file to False so requirements.txt existence check will return False
-        monkeypatch.setattr(Path, "is_file", lambda *args, **kwargs: False)
-        monkeypatch.setattr(cloud, "Path", Path)
-
         # This is the main assertion:
         # we have an existing instance on `cluster-001`
         # but we want to run this app on `cluster-002`
-        cloud_runtime.dispatch(name=app_name, cluster_id=new_cluster)
-
-        body = Body8(
-            cluster_id=new_cluster,
-            app_entrypoint_file=mock.ANY,
-            enable_app_server=True,
-            flow_servers=[],
-            image_spec=None,
-            works=[],
-            local_source=True,
-            dependency_cache_key=mock.ANY,
-            user_requested_flow_compute_config=mock.ANY,
-        )
-        cloud_runtime.backend.client.lightningapp_v2_service_create_lightningapp_release.assert_called_once_with(
-            project_id="default-project-id", app_id=mock.ANY, body=body
-        )
-        cloud_runtime.backend.client.projects_service_create_project_cluster_binding.assert_called_once_with(
-            project_id="default-project-id",
-            body=V1ProjectClusterBinding(cluster_id=new_cluster, project_id="default-project-id"),
-        )
+        with expected_raise as exception:
+            cloud_runtime.dispatch(name=app_name, cluster_id=new_cluster)
+
+        if exception is not None:
+            old_cluster = old_cluster or DEFAULT_CLUSTER
+            new_cluster = new_cluster or DEFAULT_CLUSTER
+            assert str(exception.value) == dedent(
+                f"""\
+                Can not run app '{app_name}' on cluster {new_cluster} since it already exists on {old_cluster}
+                    (moving apps between clusters is not supported).
+
+                You can either:
+                    a. rename the app to run on {new_cluster} with the --name option
+                        lightning run app script.py --name (new name) --cloud --cluster-id {new_cluster}
+                    b. delete the app running on {old_cluster} in the UI before running this command.
```

Comment:
What exactly is the reason for duplicating the entire error message here? Let's not do this. Let's use the pytest match in the context manager:

```py
with pytest.raises(
    ValueError, 
    match="Can not run app '{app_name}' on cluster {new_cluster} since it already exists on {old_cluster}"
)
```
That should be enough to identify the message uniquely!